flowmachine/flowmachine/core/query.py

```diff
@@ -188,6 +193,10 @@ def get_query(self):
             schema, name = table_name.split(".")
             with rlock(self.redis, self.md5):
                 if self.connection.has_table(schema=schema, name=name):
+                    try:
+                        touch_cache(self.connection, self.md5)
+                    except ValueError:
+                        pass  # Cache record not written yet
```

Comment:
It took me a minute to understand why this exception is ok to ignore. Without re-reading the definition of `touch_cache()` I assumed it would create a cache record if it didn't exist yet (similar to how the Unix command `touch` creates an empty file if it doesn't exist). Is it worth actually making this an optional (or required?) argument, e.g. `touch_cache(self.connection, self.md5, ignore_if_not_exists=True)`

Regarding the context of this addition, why does `get_query()` want to touch the cache at all? Is it because something else is trying to make use of this query? Does it make a difference whether `get_query()` is called by some other flowmachine-internal class or by the user?

Reply:
Added a comment to explain the pass.

The equivalent to this used to happen in `is_stored`, but I felt like `get_query` is actually a better indicator of the cache record being _used_, since in most cases the only time `get_query` is called is if you're running the query to get the dataframe, or another query is including its results.

Ideally, we'd do this update with a trigger on the actual cache table being read via a SELECTm but unfortunately Postgres doesn't support that at this time.